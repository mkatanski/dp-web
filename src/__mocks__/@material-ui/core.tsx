/* eslint-disable @typescript-eslint/no-explicit-any */
import React from "react";
export * from "@material-ui/core";

export const Table = ({ children, className, ...rest }: any) => (
  <table id="MUI.Table" className={className} data-props={rest}>
    {children}
  </table>
);

export const TableBody = ({ children, ...rest }: any) => (
  <tbody id="MUI.TableBody" data-props={rest}>
    {children}
  </tbody>
);

export const TableHead = ({ children, ...rest }: any) => (
  <th id="MUI.TableHead" data-props={rest}>
    {children}
  </th>
);

export const TableContainer = ({ children, className, ...rest }: any) => (
  <div id="MUI.TableContainer" className={className} data-props={rest}>
    {children}
  </div>
);

export const TableCell = ({
  children,
  onClick,
  className,
  ...rest
}: React.PropsWithChildren<{ onClick?: () => void; className?: string }>) => (
  <td
    id="MUI.TableCell"
    onClick={onClick}
    className={className}
    data-props={rest}
  >
    {children}
  </td>
);

export const TableRow = ({ children, className, onClick, ...rest }: any) => (
  <tr
    id="MUI.TableRow"
    className={className}
    onClick={onClick}
    {...(Object.keys(rest).length ? { "data-props": { ...rest } } : {})}
  >
    {children}
  </tr>
);

export const TableSortLabel = ({ children, ...rest }: any) => (
  <div id="MUI.TableSortLabel" data-props={rest}>
    {children}
  </div>
);

export const TextField = ({ children, className, ...rest }: any) => (
  <input id="MUI.TextField" className={className} data-props={rest}>
    {children}
  </input>
);

export const Button = ({ children, onClick, className, ...rest }: any) => (
  <button
    id="MUI.Button"
    onClick={onClick}
    className={className}
    data-testid={rest["data-testid"]}
    data-props={rest}
  >
    {children}
  </button>
);

export const Paper = ({ children, className, ...rest }: any) => (
  <div id="MUI.Paper" className={className} data-props={rest}>
    {children}
  </div>
);

export const Container = ({ children, className, ...rest }: any) => (
  <div id="MUI.Container" className={className} data-props={rest}>
    {children}
  </div>
);

export const Card = ({ children, className, ...rest }: any) => (
  <div id="MUI.Card" className={className} data-props={rest}>
    {children}
  </div>
);

export const Typography = ({ children, className, ...rest }: any) => (
  <div id="MUI.Typography" className={className} data-props={rest}>
    {children}
  </div>
);

export const Grid = ({ children, className, ...rest }: any) => (
  <div id="MUI.Grid" className={className} data-props={rest}>
    {children}
  </div>
);

export const CardContent = ({ children, className, ...rest }: any) => (
  <div id="MUI.CardContent" className={className} data-props={rest}>
    {children}
  </div>
);

export const CardHeader = ({ children, className, ...rest }: any) => (
  <div id="MUI.CardHeader" className={className} data-props={rest}>
    {children}
  </div>
);

export const Divider = ({ children, className, ...rest }: any) => (
  <div id="MUI.Divider" className={className} data-props={rest}>
    {children}
  </div>
);

export const IconButton = ({ children, onClick, className, ...rest }: any) => (
  <span id="MUI.IconButton" onClick={onClick} className={className} {...rest}>
    {children}
  </span>
);

export const Drawer = ({ children, className, ...rest }: any) => (
  <div id="MUI.Drawer" className={className} data-props={rest}>
    {children}
  </div>
);
export const TablePagination = ({ children, ...rest }: any) => (
  <div id="MUI.TablePagination" data-props={rest}>
    {children}
  </div>
);

export const List = ({ children, className, ...rest }: any) => (
  <ul id="MUI.List" className={className} data-props={rest}>
    {children}
  </ul>
);

export const ListSubheader = ({ children, className, ...rest }: any) => (
  <li id="MUI.ListSubheader" className={className} data-props={rest}>
    {children}
  </li>
);

export const ListItem = ({ children, className, ...rest }: any) => (
  <li id="MUI.ListItem" className={className} data-props={rest}>
    {children}
  </li>
);

export const CardActions = ({ children, className, ...rest }: any) => (
  <div id="MUI.CardActions" className={className} data-props={rest}>
    {children}
  </div>
);

export const AppBar = ({ children, ...rest }: any) => (
  <div id="MUI.AppBar" data-props={rest}>
    {children}
  </div>
);

export const Toolbar = ({ children, ...rest }: any) => (
  <div id="MUI.Toolbar" data-props={rest}>
    {children}
  </div>
);

export const CircularProgress = ({ className, ...rest }: any) => (
  <div id="MUI.CircularProgress" className={className} data-props={rest} />
);

export const LinearProgress = ({ className, ...rest }: any) => (
  <div id="MUI.LinearProgress" className={className} data-props={rest} />
);

export const Box = ({ children, className, ...rest }: any) => (
  <div id="MUI.Box" className={className} data-props={rest}>
    {children}
  </div>
);

export const Switch = ({ ...rest }: any) => (
  <div id="MUI.Switch" data-props={rest} />
);

export const FormControl = ({ children, className, ...rest }: any) => (
  <div id="MUI.FormControl" className={className} data-props={rest}>
    {children}
  </div>
);

export const Select = ({ children, className, name, ...rest }: any) => (
  <select id="MUI.Select" className={className} name={name} data-props={rest}>
    {children}
  </select>
);

export const FormHelperText = ({ children, className, ...rest }: any) => (
  <div id="MUI.FormHelperText" className={className} data-props={rest}>
    {children}
  </div>
);

export const InputLabel = ({ children, className, ...rest }: any) => (
  <div id="MUI.InputLabel" className={className} data-props={rest}>
    {children}
  </div>
);

export const MenuItem = ({ children, className, ...rest }: any) => (
  <option id="MUI.MenuItem" className={className} data-props={rest}>
    {children}
  </option>
);

export const FormGroup = ({ children, className, ...rest }: any) => (
  <div id="MUI.FormGroup" className={className} data-props={rest}>
    {children}
  </div>
);

export const FormControlLabel = ({ control, className, ...rest }: any) => (
  <div id="MUI.FormControlLabel" className={className} data-props={rest}>
    {control}
  </div>
);

export const ListItemAvatar = ({ children, className, ...rest }: any) => (
  <div id="MUI.ListItemAvatar" className={className} data-props={rest}>
    {children}
  </div>
);

export const Tabs = ({ children, className, ...rest }: any) => (
  <div id="MUI.Tabs" className={className} data-props={rest}>
    {children}
  </div>
);

export const Avatar = ({ className, ...rest }: any) => (
  <img id="MUI.Avatar" className={className} data-props={rest} />
);

export const Tab = ({ children, className, ...rest }: any) => (
  <div id="MUI.Tab" className={className} data-props={rest}>
    {children}
  </div>
);

export const Collapse = ({ children, className, ...rest }: any) => (
  <div id="MUI.Collapse" className={className} data-props={rest}>
    {children}
  </div>
);

export const Chip = ({ children, className, ...rest }: any) => (
  <div id="MUI.Chip" className={className} data-props={rest}>
    {children}
  </div>
);

export const Menu = ({ children, onClick, className, ...rest }: any) => (
  <div id="MUI.Menu" onClick={onClick} className={className} data-props={rest}>
    {children}
  </div>
);
