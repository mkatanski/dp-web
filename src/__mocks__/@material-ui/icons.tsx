/* eslint-disable @typescript-eslint/no-explicit-any */
import React from "react";
export * from "@material-ui/icons";

export const ShowChart = ({ ...rest }: any) => (
  <span id="MUI.Icon.ShowChart" data-props={rest} />
);

export const Assignment = ({ ...rest }: any) => (
  <span id="MUI.Icon.Assignment" data-props={rest} />
);

export const AssignmentTurnedIn = ({ ...rest }: any) => (
  <span id="MUI.Icon.AssignmentTurnedIn" data-props={rest} />
);

export const Warning = ({ ...rest }: any) => (
  <span id="MUI.Icon.Warning" data-props={rest} />
);

export const HomeOutlined = ({ ...rest }: any) => (
  <span id="MUI.Icon.HomeOutlined" data-props={rest} />
);

export const StorefrontOutlined = ({ ...rest }: any) => (
  <span id="MUI.Icon.StorefrontOutlined" data-props={rest} />
);

export const InputOutlined = ({ ...rest }: any) => (
  <span id="MUI.Icon.InputOutlined" data-props={rest} />
);

export const MenuOutlined = ({ ...rest }: any) => (
  <span id="MUI.Icon.MenuOutlined" data-props={rest} />
);
export const TrendingUpOutlined = ({ ...rest }: any) => (
  <span id="MUI.Icon.MenuOutlined" data-props={rest} />
);

export const Save = ({ ...rest }: any) => (
  <span id="MUI.Icon.Save" data-props={rest} />
);

export const DeleteOutline = ({ ...rest }: any) => (
  <span id="MUI.Icon.DeleteOutline" data-props={rest} />
);

export const Add = ({ ...rest }: any) => (
  <span id="MUI.Icon.Add" data-props={rest} />
);

export const WarningOutlined = ({ ...rest }: any) => (
  <span id="MUI.Icon.WarningOutlined" data-props={rest} />
);

export const ExpandMoreOutlined = ({ ...rest }: any) => (
  <span id="MUI.Icon.ExpandMoreOutlined" data-props={rest} />
);

export const ExpandLessOutlined = ({ ...rest }: any) => (
  <span id="MUI.Icon.ExpandLessOutlined" data-props={rest} />
);

export const Dashboard = ({ ...rest }: any) => (
  <span id="MUI.Icon.Dashboard" data-props={rest} />
);
