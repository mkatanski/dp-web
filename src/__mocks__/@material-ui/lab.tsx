/* eslint-disable @typescript-eslint/no-explicit-any */
import React from "react";

export const ToggleButton = ({
  children,
  onClick,
  className,
  ...rest
}: any) => (
  <button
    id="MUI.ToggleButton"
    onClick={onClick}
    className={className}
    data-testid={rest["data-testid"]}
    data-props={rest}
  >
    {children}
  </button>
);

export const ToggleButtonGroup = ({
  children,
  onClick,
  className,
  ...rest
}: any) => (
  <div
    id="MUI.ToggleButtonGroup"
    onClick={onClick}
    className={className}
    data-testid={rest["data-testid"]}
    data-props={rest}
  >
    {children}
  </div>
);

export const Skeleton = ({ ...rest }: any) => (
  <div id="MUI.Skeleton" data-props={rest} />
);
