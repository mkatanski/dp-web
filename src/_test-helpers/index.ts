export * from "./withResponsiveContext";
export * from "./withThemeProvider";
