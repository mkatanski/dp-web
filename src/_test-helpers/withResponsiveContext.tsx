import React from "react";
import { Context as ResponsiveContext } from "react-responsive";
import { Breakpoints } from "config/breakpoints";

export const withMobileSize = (Component: React.ReactNode) => (
  <ResponsiveContext.Provider value={{ width: Breakpoints.mobile }}>
    {Component}
  </ResponsiveContext.Provider>
);

export const withTabletSize = (Component: React.ReactNode) => (
  <ResponsiveContext.Provider value={{ width: Breakpoints.tablet }}>
    {Component}
  </ResponsiveContext.Provider>
);

export const withDesktopSize = (Component: React.ReactNode) => (
  <ResponsiveContext.Provider value={{ width: Breakpoints.desktop }}>
    {Component}
  </ResponsiveContext.Provider>
);

export const withLargeDesktopSize = (Component: React.ReactNode) => (
  <ResponsiveContext.Provider value={{ width: Breakpoints.largeDesktop }}>
    {Component}
  </ResponsiveContext.Provider>
);
