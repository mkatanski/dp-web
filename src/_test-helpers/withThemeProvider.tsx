import React from "react";
import { ThemeProvider as DefaultThemeProvider } from "lib/theme/ThemeProvider";
import { StyledTheme, Theme } from "styles/theme/_types";
import { createMuiTheme } from "@material-ui/core";

const testTheme: StyledTheme = {
  primary: "primary",
  secondary: "secondary",
  textPrimary: "textPrimary",
  textSecondary: "textSecondary",
  textInverse: "textInverse"
};

const muiTheme = createMuiTheme({
  direction: "ltr"
});

const theme: Theme = Object.assign(testTheme, muiTheme);

export const withThemeProvider = (Component: React.ReactNode) => (
  <DefaultThemeProvider theme={theme}>{Component}</DefaultThemeProvider>
);
