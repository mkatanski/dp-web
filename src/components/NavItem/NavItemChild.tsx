import React from "react";
import { NavLink as RouterLink } from "react-router-dom";
import { Button, ListItem, SvgIconProps } from "@material-ui/core";
import styled from "styled-components";

const DEFAULT_SPACING = 8;
const LINK_INDENT = 32;

export type NavItemChildProps = {
  title: string;
  href: string;
  depth: number;
  icon?: React.ReactType<SvgIconProps>;
  info?: React.ReactNode;
};

const ListItemWrapper = styled.span`
  .itemLeaf {
    display: flex;
    padding-top: 0;
    padding-bottom: 0;
  }

  .linkButtonLeaf {
    color: ${({ theme }) => theme.palette.text.secondary};
    padding: ${({ theme }) => theme.spacing(1.25, 1)};
    justify-content: flex-start;
    text-transform: none;
    letter-spacing: 0;
    width: 100%;
    font-weight: ${({ theme }) => theme.typography.fontWeightRegular};

    &.depth-0 {
      &.title {
        font-weight: ${({ theme }) => theme.typography.fontWeightMedium};
      }
    }
  }

  .linkTitle {
    margin-right: auto;
  }

  .icon {
    display: flex;
    align-items: center;
    margin-right: ${({ theme }) => `${theme.spacing(1)}px`};
  }

  .linkActive {
    color: ${({ theme }) => theme.palette.secondary.main};

    .linkTitle {
      font-weight: ${({ theme }) => theme.typography.fontWeightMedium};
    }

    .icon {
      color: ${({ theme }) => theme.palette.secondary.main};
    }
  }
`;

export const NavItemChild: React.FC<NavItemChildProps> = ({
  title,
  href,
  depth,
  icon: Icon,
  info: Info
}: NavItemChildProps) => {
  let paddingLeft = DEFAULT_SPACING;

  depth > 0 && (paddingLeft = LINK_INDENT + DEFAULT_SPACING * depth);

  const style = { paddingLeft };

  return (
    <ListItemWrapper>
      <ListItem className="itemLeaf" disableGutters key={title}>
        <Button
          component={RouterLink}
          activeClassName="linkActive"
          className={`linkButtonLeaf depth-${depth}`}
          style={style}
          to={href}
        >
          {Icon && <Icon className="icon" fontSize="default" />}
          <span className="linkTitle">{title}</span>
          {Info}
        </Button>
      </ListItem>
    </ListItemWrapper>
  );
};
