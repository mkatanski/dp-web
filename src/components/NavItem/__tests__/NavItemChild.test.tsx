/* eslint-disable react/display-name */
import React from "react";
import TestRenderer from "react-test-renderer";
import "jest-styled-components";

import { withThemeProvider } from "_test-helpers";
import { StorefrontOutlined } from "@material-ui/icons";
import { NavItemChild } from "../NavItemChild";

describe("components/NavItem", () => {
  describe("NavItemChild", () => {
    it("should render NavItemChild component", () => {
      const tree = TestRenderer.create(
        withThemeProvider(
          <NavItemChild
            title="Products"
            href="/products"
            depth={1}
            icon={StorefrontOutlined}
          />
        )
      );

      expect(tree).toMatchSnapshot();
    });
  });
});
