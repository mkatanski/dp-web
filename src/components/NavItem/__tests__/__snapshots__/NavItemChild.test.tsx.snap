// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/NavItem NavItemChild should render NavItemChild component 1`] = `
.c0 .itemLeaf {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-top: 0;
  padding-bottom: 0;
}

.c0 .linkButtonLeaf {
  color: #546e7a;
  padding: 10px 8px;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  text-transform: none;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  width: 100%;
  font-weight: 400;
}

.c0 .linkButtonLeaf.depth-0.title {
  font-weight: 500;
}

.c0 .linkTitle {
  margin-right: auto;
}

.c0 .icon {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: 8px;
}

.c0 .linkActive {
  color: #5850EC;
}

.c0 .linkActive .linkTitle {
  font-weight: 500;
}

.c0 .linkActive .icon {
  color: #5850EC;
}

<span
  className="c0"
>
  <li
    className="itemLeaf"
    data-props={
      Object {
        "disableGutters": true,
      }
    }
    id="MUI.ListItem"
  >
    <button
      className="linkButtonLeaf depth-1"
      data-props={
        Object {
          "activeClassName": "linkActive",
          "component": [Function],
          "style": Object {
            "paddingLeft": 40,
          },
          "to": "/products",
        }
      }
      id="MUI.Button"
    >
      <span
        data-props={
          Object {
            "className": "icon",
            "fontSize": "default",
          }
        }
        id="MUI.Icon.StorefrontOutlined"
      />
      <span
        className="linkTitle"
      >
        Products
      </span>
    </button>
  </li>
</span>
`;
